packages/viewer/src/components/viewer-transform-widget/util.ts

```diff
@@ -67,6 +82,89 @@ export function convertCanvasPointToWorld(
   return undefined;
 }
 
+export function computeInputTransform(
+  identifier: string,
+  value: number,
+  lastValue: number,
+  distanceUnit: DistanceUnitType = 'millimeters',
+  angleUnit: AngleUnitType = 'degrees'
+): Matrix4.Matrix4 {
+  const units = new DistanceUnits(distanceUnit);
+  const angles = new AngleUnits(angleUnit);
+
+  const rotation = (): number => angles.convertFrom(value - lastValue);
+  const position = (): number =>
+    units.convertRealValueToWorld(value - lastValue);
+
+  switch (identifier) {
+    case 'x-translate':
+      return Matrix4.makeTranslation(Vector3.create(position(), 0, 0));
+    case 'y-translate':
+      return Matrix4.makeTranslation(Vector3.create(0, position(), 0));
+    case 'z-translate':
+      return Matrix4.makeTranslation(Vector3.create(0, 0, position()));
+    case 'x-rotate':
+      return Matrix4.makeRotation(
+        Quaternion.fromAxisAngle(Vector3.left(), rotation())
+      );
+    case 'y-rotate':
+      return Matrix4.makeRotation(
+        Quaternion.fromAxisAngle(Vector3.down(), rotation())
+      );
+    case 'z-rotate':
+      return Matrix4.makeRotation(
+        Quaternion.fromAxisAngle(Vector3.forward(), rotation())
+      );
+  }
+  return Matrix4.makeIdentity();
+}
+
+export function computeInputDisplayValue(
+  identifier: string,
+  current: Matrix4.Matrix4,
+  start: Matrix4.Matrix4,
+  distanceUnit: DistanceUnitType = 'millimeters',
```

Comment:
I don't feel too strongly about making a `DefaultUnitType` here, since the `DistanceUnitType` definition looks like 
```typescript
export type DistanceUnitType =
  // Metric
  | 'millimeters'
  | 'centimeters'
  | 'meters'

  // Imperial
  | 'inches'
  | 'feet'
  | 'yards';
```
such that the `'millimeters'` string has type safety. That said, we do also default the top-level unit to millimeters, so this default is probably redundant. I'll remove it!